### PR TITLE
Add types to `Package` tests

### PR DESCRIPTION
**Pull request by herndlm**
Contribution to #10159

### CODE REVIEW

File: tests/Composer/Test/Package/Archiver/PharArchiverTest.php [hunk 1]
```diff
@@ -63,13 +65,20 @@ protected function setupDummyRepo()
...
+     * @return void
+     */
     protected function writeFile($path, $content, $currentWorkDir)
     {
         if (!file_exists(dirname($path))) {
             mkdir(dirname($path), 0777, true);
         }
 
-        $result = file_put_contents($path, 'a');
+        $result = file_put_contents($path, $content);
```
Comment (herndlm) [inline tests/Composer/Test/Package/Archiver/PharArchiverTest.php:81]: Reads like an irrelevant bug/typo, `$content` was unused here as it was only ever writing 'a' and not read back anyways, so the content is not important. But this makes the method look like exactly the one in `ZipArchiverTest`.

File: tests/Composer/Test/Package/LockerTest.php [hunk 2]
```diff
@@ -255,13 +255,19 @@ public function testIsFreshFalseWithContentHash()
         $this->assertFalse($locker->isFresh());
     }
 
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\Composer\Json\JsonFile
```
Comment (Seldaek) [inline tests/Composer/Test/Package/LockerTest.php:259]:
```suggestion
     * @return \PHPUnit\Framework\MockObject\MockObject&\Composer\Json\JsonFile
```

File: tests/Composer/Test/Package/LockerTest.php [hunk 3]
```diff
@@ -255,13 +255,19 @@ public function testIsFreshFalseWithContentHash()
...
+     */
     private function createJsonFileMock()
     {
         return $this->getMockBuilder('Composer\Json\JsonFile')
             ->disableOriginalConstructor()
             ->getMock();
     }
 
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\Composer\Installer\InstallationManager
```
Comment (Seldaek) [inline tests/Composer/Test/Package/LockerTest.php:269]:
```suggestion
     * @return \PHPUnit\Framework\MockObject\MockObject&\Composer\Installer\InstallationManager
```

File: tests/Composer/Test/Package/LockerTest.php [hunk 4]
```diff
@@ -271,12 +277,20 @@ private function createInstallationManagerMock()
         return $mock;
     }
 
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\Composer\Package\PackageInterface
```
Comment (Seldaek) [inline tests/Composer/Test/Package/LockerTest.php:281]:
```suggestion
     * @return \PHPUnit\Framework\MockObject\MockObject&\Composer\Package\PackageInterface
```

File: tests/Composer/Test/Package/Version/VersionSelectorTest.php [hunk 5]
```diff
@@ -353,13 +358,21 @@ public function getRecommendedRequireVersionPackages()
...
+     */
     private function createPackage($version)
     {
         $parser = new VersionParser();
 
         return new Package('foo', $parser->normalize($version), $version);
     }
 
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\Composer\Repository\RepositorySet
```
Comment (Seldaek) [inline tests/Composer/Test/Package/Version/VersionSelectorTest.php:374]:
```suggestion
     * @return \PHPUnit\Framework\MockObject\MockObject&\Composer\Repository\RepositorySet
```

**Review by Seldaek — APPROVED** [review]

**Comment by Seldaek** [comment]
Thanks!

**Comment by herndlm** [comment]
Oh, thank you for the fixes 😅

**Pull request merged**